### PR TITLE
refactor(experimental): follow creator pattern for airdrop helper

### PR DESCRIPTION
**Pull request by steveluscher**
# Summary

I regret not applying the factory pattern for the airdrop helper as well. This new design will allow other devs to create their _own_ airdrop helpers with custom RPC and transaction confirimation strategies, should they wish to. Everyone else can use the default RPCs and create a default airdrop helper.

# Test Plan

```
cd packages/library
pnpm test:unit:browser
pnpm test:unit:node
```

Related to #1579.

### CODE REVIEW

**Comment by steveluscher** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1754** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1754" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1754?utm_source=stack-comment).

File: packages/library/src/__tests__/airdrop-test.ts [hunk 1]
```diff
@@ -2,14 +2,10 @@ import { Base58EncodedAddress } from '@solana/addresses';
...
 import { RequestAirdropApi } from '@solana/rpc-core/dist/types/rpc-methods/requestAirdrop';
-import { SignatureNotificationsApi } from '@solana/rpc-core/dist/types/rpc-subscriptions/signature-notifications';
-import { Rpc, RpcSubscriptions } from '@solana/rpc-transport/dist/types/json-rpc-types';
+import { Rpc } from '@solana/rpc-transport/dist/types/json-rpc-types';
 import { TransactionSignature } from '@solana/transactions';
 
 import { requestAndConfirmAirdrop } from '../airdrop';
-import { createDefaultSignatureOnlyRecentTransactionConfirmer } from '../airdrop-confirmer';
-
-jest.mock('../airdrop-confirmer');
```
Comment (steveluscher) [inline packages/library/src/__tests__/airdrop-test.ts:12]: This is where I should have known that the other design was wrong.

**Review by buffalojoec — APPROVED** [review]
Lgtm

**Review by mcintyre94 — APPROVED** [review]
Nice, looks great!

**Pull request merged**